taichi/program/async_engine.cpp

```diff
@@ -281,6 +282,16 @@ TaskMeta AsyncEngine::create_task_meta(
     }
     return false;
   });
+  if (root_stmt->task_type == OffloadedStmt::listgen) {
+    meta.input_states.emplace_back(root_stmt->snode, AsyncState::Type::list);
+    meta.input_states.emplace_back(root_stmt->snode, AsyncState::Type::mask);
+    meta.output_states.emplace_back(root_stmt->snode, AsyncState::Type::list);
+  } else if (root_stmt->task_type == OffloadedStmt::struct_for) {
+    meta.input_states.emplace_back(root_stmt->snode, AsyncState::Type::list);
```

Comment:
Once again, I cannot comment on lines that are not modified... For write ops like `GlobalStoreStmt` and `AtomicOpStmt`, if they are operating on `SNode`, should we also add `mask` state to this meta's `output_states`?

Reply:
Yes, you are right! Thanks for pointing this out :-)